src/app/network/UploadManager.ts

```diff
@@ -110,16 +113,33 @@ class UploadManager {
         },
       });
 
+      const existsRelatedTask = !!this.options?.relatedTaskId;
+      if (!existsRelatedTask) localStorageService.setUploadState(taskId, TaskStatus.InProcess);
+
+      const retyUploadType = fileData.fileType;
+
       const upload = () => {
         uploadAttempts++;
         const isMultipleUpload = this.items.length > 1 ? 1 : 0;
 
+        if (!existsRelatedTask)
+          tasksService.updateTask({
+            taskId: taskId,
+            merge: {
+              status: TaskStatus.InProcess,
+            },
+          });
+
+        const uploadStatus = localStorageService.getUploadState(this.options?.relatedTaskId ?? taskId);
```

Comment:
The upload manager should not know anything about the localStorage, this is a very coupled implementation. I would need to decouple this first if I want to test only the manager behavior